elfi/post_proc.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from collections import OrderedDict
+import inspect
+
+import elfi.visualization as vis
+
+
+"""
+Implementations for post-processing.
+"""
+
+
+class Result(object):
+    """Container for results from ABC methods. Allows intuitive syntax for plotting etc.
+
+    Parameters
+    ----------
+    samples : OrderedDict of name: np.arrays
+    """
+    def __init__(self, samples, **kwargs):
+        self.samples = samples
+        self.n_samples = len(list(self.samples.values())[0])
+        self.n_params = len(samples)
+
+        # get name of the ABC method
+        stack10 = inspect.stack()[1][0]
+        self.method = stack10.f_locals["self"].__class__.__name__
```

Comment:
I'm not absolutely sure if this works in all cases, but if it does, why give an extra argument?